runtime/jaxrs/src/main/java/com/flit/runtime/jaxrs/FlitExceptionMapper.java

```diff
@@ -0,0 +1,40 @@
+package com.flit.runtime.jaxrs;
+
+import com.flit.runtime.FlitException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Provider
+public class FlitExceptionMapper implements ExceptionMapper<FlitException> {
```

Comment:
Do we still need these `ExceptionMapper`s? 
I know that we're going to need some `ExceptionMapper` fir Dropwizard, but does it still make sense to have them in Flit or have the app build them as needed? 

I can see the `InvalidProtobufExceptionMapper` being useful, but Idk about the `FlitExceptionMapper` as the generate code doesn't throw any `FlitException`s.


Also does it make sense to put Dropwizard only code in flit now? I believe that the the`ExceptionMapper` currently only work with Dropwizard as I don't see any `@ExceptionHandler(...` annotations for springboot.

Reply:
@ShaneSaww I think we still need them. We could definitely move them over to fjord but the existing pattern is to declare the exception mappers for each server type (spring mvc, undertow and in this case jaxrs). 

> I can see the InvalidProtobufExceptionMapper being useful, but Idk about the FlitExceptionMapper as the generate code doesn't throw any FlitExceptions.

So the usefulness of these `ExceptionMappers` is to convert the FlitExceptions to the appropriate response. The application, in this case Fjord, would have to throw `FlitException` wrapping existing exceptions for this mapper to catch and convert to the response. In aqueduct, we actually wrap the exception to a `FlitException` [here](https://github.com/github/aqueduct/blob/master/src/main/java/com/github/aqueduct/core/twirp/JobQueueService.java#L292). The only difference is that Spring's exception handler covers both the mappers (`Exception` and `InvalidProtocolException`) in one class. 

> Also does it make sense to put Dropwizard only code in flit now? I believe that the theExceptionMapper currently only work with Dropwizard as I don't see any @ExceptionHandler(... annotations for springboot.

So if you look at the project outline, there are 3 packages under runtime:

- jaxrs
- spring
- undertow

`ExceptionMapper` is a JAX-RS concept and will work in any of the frameworks (dropwizard, spring, undertow) as long as you are using the JAX-RS library instead of like Spring Rest/MVC. The equivalent of the `ExceptionMapper` for spring is [FlitExceptionHandler](https://github.com/github/flit/blob/master/runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java), which we currently use.

Reply:
>The application, in this case Fjord, would have to throw FlitException wrapping existing exceptions for this mapper to catch and convert to the response. In aqueduct, we actually wrap the exception to a FlitException here. 

ohh, interesting. I fully didn't get that before. Just thinking about it, it makes some sense, but it's also a bit weird to me.  It seems like a lot of work to get the correct response code(and error message) back to the client. But nothing we need to change right now. 

>So if you look at the project outline, there are 3 packages under runtime:

👍  I completely missed the spring runtime one. 

>ExceptionMapper is a JAX-RS concept and will work in any of the frameworks (dropwizard, spring, undertow) as long as you are using the JAX-RS library instead of like Spring Rest/MVC. The equivalent of the ExceptionMapper for spring is FlitExceptionHandler, which we currently use.

👍  Thanks for the explanation